vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/reasoner/Reasoner.java

```diff
@@ -331,6 +331,67 @@ static Reasoner getInstance() {
 	 */
 	QueryResultIterator answerQuery(PositiveLiteral query, boolean includeNulls);
 
+	/*
+	 * @param query a {@link PositiveLiteral} representing the query to be answered.
+	 *
+	 * @return queryAnswerSize(query, true), the number of facts in the extension of
+	 * the query.
+	 */
+	long queryAnswerSize(PositiveLiteral query);
+
+	// TODO add examples to query javadoc
+	/**
+	 * Evaluates an atomic ({@code query}), and returns the number of implicit facts
+	 * loaded into the reasoner and the number of  explicit facts materialised by
+	 * the reasoner.
+	 * <br>
+	 * An answer to the query is the terms a fact that matches the {@code query}:
+	 * the fact predicate is the same as the {@code query} predicate, the
+	 * {@link TermType#CONSTANT} terms of the {@code query} appear in the answer
+	 * fact at the same term position, and the {@link TermType#VARIABLE} terms of
+	 * the {@code query} are matched by terms in the fact, either named
+	 * ({@link TermType#CONSTANT}) or anonymous ({@link TermType#NAMED_NULL}). The
+	 * same variable name identifies the same term in the answer fact. <br>
+	 * 
+	 * Depending on the state of the reasoning (materialisation) and its
```

Comment:
we discuss Correctness in the javadoc, but `queryAnswerSize()` method does not return in any way its correctness result. We should either return Correctness together with the answer size, or ignore it in the javadoc, or suggest that Correctness can be found out together with the query result in `answerQuery` or `exportQueryAnswers`